pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -314,6 +319,40 @@ func (c *Cluster) Start() error {
 	return nil
 }
 
+func (c *Cluster) postReconcileUpdateOSDProperties(desiredOSDs map[int]*OSDInfo) error {
+	osdUsage, err := cephclient.GetOSDUsage(c.context, c.clusterInfo)
+	if err != nil {
+		return errors.Wrap(err, "failed to get osd usage")
+	}
+	logger.Infof("post processing osd properties with %d actual osds from ceph osd df and %d existing osds found during reconcile", len(osdUsage.OSDNodes), len(desiredOSDs))
+	for _, actualOSD := range osdUsage.OSDNodes {
+		if desiredOSD, ok := desiredOSDs[actualOSD.ID]; ok {
+			if err := c.updateDeviceClassIfChanged(*desiredOSD, actualOSD); err != nil {
+				// Log the error and allow other updates to continue
+				logger.Error(err)
+			}
+		}
+	}
+	return nil
+}
+
+func (c *Cluster) updateDeviceClassIfChanged(desired OSDInfo, actual cephclient.OSDNodeUsage) error {
+	if !c.spec.Storage.AllowDeviceClassUpdate {
+		// device class updates are not allowed by default
+		return nil
+	}
+	if desired.DeviceClass != "" && desired.DeviceClass != actual.DeviceClass {
```

Comment:
```suggestion
if desired.DeviceClass != "" && desired.DeviceClass == actual.DeviceClass {
logger.Infof("no device class change needed for osd.%d. desired=%q, actual=%q", desired.ID, desired.DeviceClass, actual.DeviceClass)
return nil
}
```

How about if we return early we don't need to have else statement.